### PR TITLE
Maximal Square Solution (Leetcode) Added problem number 221

### PR DESCRIPTION
**Pull request by vivekvardhan2810**
## Related Issue

Fixed : #843

## Description

I had Added the Solution of problem "Minimum Cost to Connect Two Groups of Points" under Leetcode Solution > 0200 - 0299> 221

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [x] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by vivekvardhan2810** [comment]
Hi @Ajay-Dhangar , i have added solution you can review it.

And tell me the suggestions.

File: dsa-solutions/lc-solutions/0200-0299/221 - Maximal Square.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+---
+id: Maximal Square
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0200-0299/221 - Maximal Square.md:2]: replace `id: Maximal Square` to `id: maximal-square`

File: dsa-solutions/lc-solutions/0200-0299/221 - Maximal Square.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+- Dynamic Programming
+- Java
+- Matrix
+- Array
+description: "This document provides a solution where we find the largest square containing only 1's and return its area."
+---
+
+## Problem
+
+You are given an m x n binary $matrix$ filled with $0's$ and $1's$, find the largest square containing only $1's$ and return its area.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0200-0299/221 - Maximal Square.md:15]: replace `$0's$` to `0's` and `$1's$` to `1's`

File: dsa-solutions/lc-solutions/0200-0299/221 - Maximal Square.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+**Input:** matrix = [["0"]]
+
+**Output:** 0
+
+### Constraints
+
+- $m == matrix.length$
+- $n == matrix[i].length$
+- $1 <= m, n <= 300$
+- $matrix[i][j]$ $is$ $'0'$ $or$ $'1'$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0200-0299/221 - Maximal Square.md:46]: replace `$is$` to `is`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by vivekvardhan2810** [comment]
Hi, @Ajay-Dhangar resolved, can you check it and give suggestions

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**